File: Reflow/Provider/ILinkReader.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;

namespace Reflow.Provider
{
    public interface ILinkReader
    {
        void Initialize(IDataLink link);
        string Command
        {
            get;
            set;
        }

        bool Open();
        bool Open(string command);

        IDataReader Reader
        {
            get;
        }

    }
}

File: Java/src/main/java/com/nuarca/etl/provider/ILinkReader.java
package com.nuarca.etl.provider;

import java.sql.ResultSet;

public interface ILinkReader {

    void initialize(IDataLink link) throws Exception ;

    String getCommand() throws Exception ;

    void setCommand(String value) throws Exception ;

    boolean open() throws Exception ;

    boolean open(String command) throws Exception ;

    ResultSet getReader() throws Exception ;
}

